whelktool/scripts/syslib/litteraturbanken-create-links-holdings.groovy

```diff
@@ -0,0 +1,201 @@
+
+            instance["associatedMedia"].each { it ->
+                asList(it["uri"]).each { uri ->
+                    if (uri instanceof String && uri.contains("${shortUrl}")) { // kolla om uri matchar url från infilen (https:// borttaget då det kan vara sparat med http i Libris och vi vill inte skapa dubbletter)
+                    foundUrl = true
+                	}
+                    if (uri instanceof String && uri.contains("${normUrl}")) { // uri är ofta utan å,ä,ö - därför kollas också en sådan "normaliserad" version
+                    foundUrl = true
+                    }
+	
+                    if (epub) {
+                        if (uri instanceof String && uri.contains("${shortEpub}")) { // gör motsvarande för epub
+                            foundEpub = true
+                        }
+                        if (uri instanceof String && uri.contains("${normEpub}")) { 
+                            foundEpub = true
+                        }
+                    }
+                }
+            }
+        }
+
+        if (omslagsbild && !instance["isPrimaryTopicOf"]) { // lägg till isPrimaryTopicOf om det inte finns
+            instance["isPrimaryTopicOf"] = []
+        }
+        
+        if (instance["isPrimaryTopicOf"]) {
+
+            instance["isPrimaryTopicOf"].each { it ->
+                asList(it["uri"]).each { uri ->
+ 	
+                    if (omslagsbild) {
+                        if (uri instanceof String && uri.contains("${shortCover}")) { // kolla om uri matchar url för omslag från infilen, https:// borttaget
+                            foundCover = true
+                        }
+                    }
+                }
+            }
+        }
+        
+        if (!foundUsage) {
+            instance["usageAndAccessPolicy"] <<
+            [
+             "@type": "UsePolicy",
+             "label": "Rättigheter för Litteraturbankens texter",
+             "uri": ["http://litteraturbanken.se/om/rattigheter"]
+             ]
+        }
+        
+        if (!foundUrl) {
+            instance["associatedMedia"] <<
+            [
+            "uri": ["${url}"],
+            "@type": "MediaObject",
+            "cataloguersNote": ["856free"],
+            "marc:publicNote": "Fritt tillgänglig via Litteraturbankens webbplats"
+            ]
+        }
+        
+          if (epub && !foundEpub) {
+            instance["associatedMedia"] <<
+            [
+            "uri": ["${epub}"],
+            "@type": "MediaObject",
+            "marc:publicNote": "Hämta gratis epub direkt från Litteraturbanken"
+            ]
+        }
+        
+        if (omslagsbild && !foundCover) {
+            instance["isPrimaryTopicOf"] <<
+            [
+            "uri": ["${omslagsbild}"],
+            "@type": "Document",
+            "cataloguersNote": ["digipic"],
+            "marc:publicNote": "Titelsida"
+            ]
+        }
+        
+        bib.scheduleSave(loud: true, onError: { e ->
```

Comment:
This works however `bib.scheduleSave()` is sufficient, errors will be written to ERRORS.txt anyway.